### PR TITLE
Eth 584 erc677 in operator factory

### PR DESCRIPTION
**Pull request by samigabor**

### CODE REVIEW

**Review by jtakalai — APPROVED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/OperatorFactory.sol [hunk 1]
```diff
@@ -74,21 +74,25 @@ contract OperatorFactory is Initializable, UUPSUpgradeable, ERC2771ContextUpgrad
...
-    //     IERC677(tokenAddress).transferAndCall(newOperatorAddress, amount, "");
-    // }
+    function onTokenTransfer(address from, uint amount, bytes calldata param) external {
+        (
+            uint operatorsCutFraction,
+            string memory poolTokenName,
+            string memory operatorMetadataJson,
+            address[3] memory policies,
+            uint[3] memory policyParams
+        ) = abi.decode(param, (uint, string, string, address[3], uint[3]));
```
Comment (samt1803) [inline packages/network-contracts/contracts/OperatorTokenomics/OperatorFactory.sol:84]: maybe add a test for this with empty/not long enough data

**Review by samt1803 — APPROVED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/OperatorFactory.sol (hunk 1 above)
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/OperatorFactory.sol:84]: Added
